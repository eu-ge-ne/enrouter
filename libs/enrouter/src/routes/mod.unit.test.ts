import { describe, test, expect } from "vitest";

import { buildRoutes } from "./mod.js";

import type { BuildRoutesParams } from "./mod.js";

describe("buildRoutes", () => {
  test("from 0 modules", () => {
    const params: BuildRoutesParams = {
      entryId: "src/main.tsx",
      getModuleAssets: () => ({ modules: [], styles: [] }),
      modules: {},
    };

    expect(buildRoutes(params)).toMatchSnapshot();
  });

  test("from 1 module", () => {
    const params: BuildRoutesParams = {
      entryId: "src/main.tsx",
      getModuleAssets: (moduleId) => ({ modules: [moduleId], styles: [] }),
      modules: {
        "src/_layout.tsx": {
          path: "_layout.tsx",
          load: async () => undefined,
        },
      },
    };

    expect(buildRoutes(params)).toMatchSnapshot();
  });

  test("from 2 parent-child modules", () => {
    const params: BuildRoutesParams = {
      entryId: "src/main.tsx",
      getModuleAssets: (moduleId) => ({ modules: [moduleId], styles: [] }),
      modules: {
        "src/_layout.tsx": {
          path: "_layout.tsx",
          load: async () => undefined,
        },
        "src/abc/_layout.tsx": {
          path: "abc/_layout.tsx",
          load: async () => undefined,
        },
      },
    };

    expect(buildRoutes(params)).toMatchSnapshot();
  });
});
