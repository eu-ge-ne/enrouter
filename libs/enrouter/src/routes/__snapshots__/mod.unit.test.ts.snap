// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`buildRoutes > from 0 modules 1`] = `undefined`;

exports[`buildRoutes > from 1 module 1`] = `
{
  "link": [
    [],
    [
      "src/main.tsx",
      "src/_layout.tsx",
    ],
  ],
  "mod": [
    "src/_layout.tsx",
  ],
  "path": "/",
}
`;

exports[`buildRoutes > from 2 parent-child modules 1`] = `
{
  "link": [
    [],
    [
      "src/main.tsx",
      "src/_layout.tsx",
    ],
  ],
  "mod": [
    "src/_layout.tsx",
  ],
  "path": "/",
  "tree": [
    {
      "link": [
        [],
        [
          "src/abc/_layout.tsx",
        ],
      ],
      "mod": [
        "src/abc/_layout.tsx",
      ],
      "path": "/abc",
    },
  ],
}
`;
